evaluate_library_access_from_output_tsv/facilitate_going_through_dois_manually.R

```diff
@@ -1,94 +1,95 @@
+      '(y for on-campus, n for off-campus)? [y/n]'
+    ))
     
     if (
-      tolower(user_full_text_input) == 'y' ||
-      tolower(user_full_text_input) == 'n' ||
-      tolower(user_full_text_input) == 'invalid'
+      tolower(user_location_input) == 'y' ||
+      tolower(user_location_input) == 'n'
     ) {
-      dataset_to_go_through[
-        row_number,
-        column_for_date
-      ] <- as.character(Sys.Date())
-      
-      if (tolower(user_full_text_input) == 'y') {
-        dataset_to_go_through[row_number, column_for_data_entry] <- 1
-      } else if (tolower(user_full_text_input) == 'n') {
-        dataset_to_go_through[row_number, column_for_data_entry] <- 0
+      if (tolower(user_location_input) == 'y') {
+        column_for_data_entry <- 'full_text_indicator_manual_inside_campus'
+        column_for_date <- 'date_of_manual_full_text_check_inside_campus'
       } else {
-        dataset_to_go_through[row_number, column_for_data_entry] <- 'invalid'
+        column_for_data_entry <- 'full_text_indicator_manual_outside_campus'
+        column_for_date <- 'date_of_manual_full_text_check_outside_campus'
       }
       
       break  # Break out of the loop, and move on.
     } else {
-      message('Please enter y, n, or invalid. Asking again...')
+      message('Please enter y or n. Asking again...')
     }
   }
   
-  # Save the changes to the tsv:
-  write.table(
-    dataset_to_go_through,
-    file = manual_tsv_location,
-    sep = '\t',
-    na = '',
-    row.names = FALSE
-  )
+  for (row_number in which(
+    is.na(dataset_to_go_through[, column_for_data_entry])
+  )) {
+    doi_for_row <- dataset_to_go_through[row_number, 'doi']
+    
+    url_to_visit <- paste0(
+      'https://doi.org/',
+      doi_for_row
+    )
+    
+    message('Opening URL "', url_to_visit, '"...')
+    
+    utils::browseURL(url_to_visit)
+    
+    while (TRUE) {
+      user_full_text_input <- readline(
+        'Do we have full-text access to this DOI? [y/n/invalid]
+        ("invalid" = invalid DOI)'
+      )
+      
+      if (
+        tolower(user_full_text_input) == 'y' ||
+        tolower(user_full_text_input) == 'n' ||
+        tolower(user_full_text_input) == 'invalid'
+      ) {
+        dataset_to_go_through[
+          row_number,
+          column_for_date
+          ] <- as.character(Sys.Date())
+        
+        if (tolower(user_full_text_input) == 'y') {
+          dataset_to_go_through[row_number, column_for_data_entry] <- 1
+        } else if (tolower(user_full_text_input) == 'n') {
+          dataset_to_go_through[row_number, column_for_data_entry] <- 0
+        } else {
+          dataset_to_go_through[row_number, column_for_data_entry] <- 'invalid'
+        }
+        
+        break  # Break out of the loop, and move on.
+      } else {
+        message('Please enter y, n, or invalid. Asking again...')
+      }
+    }
+    
+    # Save the changes to the tsv:
+    write.table(
```

Comment:
Assuming this command is what caused all the quotes in the TSV, which I tend to dislike. `readr::write_tsv` defaults to not quote like this.

Reply:
Oh, no, I think it was LibreOffice Calc. I opened the TSV to check it, and may have pressed Ctrl+S out of habit before closing it again. I'll reset it...

Reply:
Quotes removed in 3ebe18d3e.

Reply:
It may have been both LibreOffice and `write.table`, though. I'm fine with changing the command, to be consistent.

Reply:
(Am changing the command now...)
Also, I just saw that you posted this comment yesterday. I didn't see it then; thank you for re-posting it now.

Reply:
Command switched in 58c348f4bc, and I checked the TSV after writing using it (d0967d7f and d0967d7f).